src/resolve/document.rs

```diff
@@ -0,0 +1,105 @@
+//! Result type of the resolution of a file include.
+use std::path::{Path, PathBuf};
+use super::{Origin, Source};
+
+/// Metadata about a data source.
+///
+/// Each document, either local or possible remote, is associated with a source identified by the
+/// host tuple as established by browsers. A path can be retrieved for every document so that you
+/// can refer to them even in other auxiliary files and generic programs. This operation
+/// potentially stores the data stream in a cache or temporary file. Each resource has one of the
+/// include types.
+pub struct Document {
+    source: Source,
+    resource: Include,
+}
+
+pub struct DocumentBuilder {
+    source: Source,
+}
+
+/// Typed representation of the resolved resource.
+///
+/// This is matched on by the `Generator` to call the respective appropriate handler
+#[derive(Debug, PartialEq, Eq)]
+pub enum Include {
+    Command(Command),
+    Markdown(PathBuf),
+    Image(PathBuf, Option<ImageMeta>),
```

Comment:
Thinking about it, if everything within `ImageMeta` is already optional, the `ImageMeta` itself should always be provided.